### PR TITLE
Fixes some bugs in the c code and makes the setup.py compatible with distutils

### PR DESCRIPTION
**Pull request by LaMi-**

### CODE REVIEW

**Comment by fgimian** [comment]
@nnathan would you be able to check out the C fixes in this pull request for me please? :smile:

File: setup.py [hunk 1]
```diff
@@ -1,4 +1,9 @@
-from setuptools import setup, Extension
+try:
+    from setuptools import setup, Extension, find_packages
+except:
```
Comment (fgimian) [inline setup.py]: Would you kindly switch this to `except ImportError`

File: setup.py [hunk 2]
```diff
@@ -1,4 +1,9 @@
...
+    from distutils.core import setup, Extension
+    find_packages = lambda: ['netsnmp']
```
Comment (fgimian) [inline setup.py]: Please rebase if you would be so kind, I have removed the use of `find_packages` :smile:

**Comment by nnathan** [comment]
Verified the C part of the code, fixes a few bugs from the original code base.

:+1:

**Comment by fgimian** [comment]
Thank you everyone! :smile:

**Comment by fgimian** [comment]
Just an FYI, I've reverted compatibility with distutils.  This complicates things for the `test` command and popular frameworks like Flask only use setuptools so I don't think it's necessary to support distutils.  I suggest installing setuptools :smile:

**Comment by LaMi-** [comment]
If you like to be compatible to most possible linux distributions while keeping dependencies low, you will need distutils. At least for some older, but still maintained distro releases. Like for example RedHat/CentOS or SLES.

This is something what I need to care about over here. But I am also fine with applying the patch on my own.

**Comment by fgimian** [comment]
Thanks for your feedback.  I'll soon be testing this with all the distributions I've listed in the README file and providing install instructions for all of them.  Even RHEL 5 allows for installation of setuptools via EPEL and heaps of other packages will also fail to install without it.  Keep in mind that I'm only supporting Python 2.6.x and 2.7.x so on such old distros, you'll need to install a newer version of Python anyway.

Please tell me which distribution that you're having issues with so I can consider it.

Thanks
Fotis

**Pull request merged**